packages/common-all/src/BacklinkUtils.ts

```diff
@@ -0,0 +1,59 @@
+import _ from "lodash";
+import { DLink, NotePropsMeta } from "./types";
+
+export class BacklinkUtils {
+  /**
+   * Create backlink out of link if it references another note (denoted by presence of link.to field)
+   *
+   * @param link Original link to create backlink out of
+   * @returns backlink or none if not applicable
+   */
+  static createFromDLink(
+    link: DLink
+  ): (Omit<DLink, "type"> & { type: "backlink" }) | undefined {
```

Comment:
Maybe extract an alias out of the return types. Something like `type BackDLink = Omit<DLink, "type"> & { type: "backlink" }`

Reply:
Sure, can do